tests/ui/assertions_on_constants.rs

```diff
@@ -1,4 +1,4 @@
-#![allow(non_fmt_panics, clippy::needless_bool, clippy::eq_op)]
+#![allow(non_fmt_panics, clippy::needless_bool, clippy::eq_op, clippy::bool_assert_comparison)]
```

Comment:
oh, it does!

given the sheer amount of pushback you gave me when you had to reorganize your code for linting reasons, I find it frankly astonishing that you would introduce a lint that forces **every** user of `const fn` to do the exact same thing you were complaining about having to do: rearchitect their code purely for linting reasons if they want this lint to remain useful for their code that doesn't use const assertions.

Reply:
lol, worry not, this lint needs feedback.  So i guess you point is that i should use `is_in_const_context` check in this lint so that it doesn't get triggered for const context.

Reply:
Yes, that seems like it would be sufficient, I think?